trieste/acquisition/function.py

```diff
@@ -130,6 +133,110 @@ def expected_improvement(model: ModelInterface, eta: tf.Tensor, at: QueryPoints)
     return (eta - mean) * normal.cdf(eta) + variance * normal.prob(eta)
 
 
+class MaxValueEntropySearch(SingleModelAcquisitionBuilder):
+    """
+    Builder for the max-value entropy search acquisition function (for function minimisation)
+    """
+
+    def __init__(self,search_space: SearchSpace, num_samples: int = 10, grid_size: int = 5000):
+        """
+        :param search_space: The global search space over which the Bayesian optimisation problem is defined.
+        :param num_samples: Number of sample draws of the minimal value.
+        :param grid_size: Size of random grid used to fit the gumbel distribution 
+            (recommend scaling with search space dimension).
+        """
+        self._search_space = search_space
+
+        if num_samples <= 0:
+            raise ValueError(
+                f"num_samples must be positive, got {num_samples}"
+            )
+        self._num_samples = num_samples
+
+        if grid_size <= 0:
+            raise ValueError(
+                f"grid_size must be positive, got {grid_size}"
+            )
+        self._grid_size = grid_size
+
+    def prepare_acquisition_function(
+        self, dataset: Dataset, model: ModelInterface
+    ) -> AcquisitionFunction:
+        """
+        Need to sample possible min-values from our posterior. 
+        To do this we implement a Gumbel sampler.
+        We approximate Pr(y*^hat<y) by Gumbel(a,b) then sample from Gumbel.
+
+        :param dataset: The data from the observer.
+        :param model: The model over the specified ``dataset``.
+        :return: The MES function.
+        """
+        query_points = self._search_space.sample(self._grid_size)
+        query_points = tf.concat([dataset.query_points,query_points],0)
+        fmean, fvar = model.predict(query_points)
+        fsd = tf.math.sqrt(fvar)
+
+        def probf(x: tf.Tensor) -> tf.Tensor :  # Build empirical CDF
+            unit_normal = tfp.distributions.Normal(tf.cast(0, fmean.dtype), tf.cast(1, fmean.dtype))
+            log_cdf = unit_normal.log_cdf(- (x - fmean) / fsd)
+            return tf.exp(tf.reduce_sum(log_cdf, axis=0))
+
+        left = tf.reduce_min(fmean - 5 * fsd)
+        right = tf.reduce_max(fmean + 5 * fsd)
+
+        def binary_search(val: float) -> float:  # Fit Gumbel quantiles
+            return bisect(lambda x: probf(x) - val, left, right, maxiter=10000, xtol=0.00001)
+        
+        q1, med, q2 = map(binary_search, [0.25, 0.5, 0.75])
+
+        b = (q1 - q2) / (tf.math.log(tf.math.log(4. / 3.)) - tf.math.log(tf.math.log(4.)))
+        a = med + b * tf.math.log(tf.math.log(2.))
+
+        uniform_samples = tf.random.uniform([self._num_samples],dtype=tf.dtypes.float64)
```

Comment:
just one missed
```suggestion
        uniform_samples = tf.random.uniform([self._num_samples],dtype=fmean.dtype)
```
and we're good to merge